### ISSUE TITLE
JavaScript rule `useless-assign` reports false positive with require and object destructuring

### ISSUE THREAD

**Issue by aarongoldenthal** [returntocorp/sgrep-rules#2862]
Update: the bug is now tracked at https://github.com/returntocorp/semgrep/issues/7531

**Describe the bug**

JavaScript rule [useless-assign](https://github.com/returntocorp/semgrep-rules/blob/develop/javascript/lang/correctness/useless-assign.yaml) reports a false positive for sequential statements with object destructuring and at least one with a `require` statement, specifically an error with assigning a value to `const` twice. I'm reporting it here because it was found with one specific rule, but it does appear to be a parsing error rather than a rule error.

The following cases all report the warnings indicated:

```js
// Message: `const` is assigned twice; the first assignment is useless
// Same error occurs wth `var` or `let` instead of `const`
const { test, expect } = require("@playwright/test");
const { pathToFileURL } = require("url");
```

```js
// Message: `const` is assigned twice; the first assignment is useless
const { test, expect } = require("@playwright/test");
const { pathToFileURL } = url;
```

```js
// Message: `const` is assigned twice; the first assignment is useless
const { test, expect } = playwright;
const { pathToFileURL } = require("url");
```

The following cases report no warning:

```js
const playwright = require("@playwright/test");
const url = require("url");
const { test, expect } = playwright;
const { pathToFileURL } = url;
```

```js
const test = require("@playwright/test").test;
const expect = require("@playwright/test").expect;
const pathToFileURL = require("url").pathToFileURL;
```

```js
const playwright = require("@playwright/test");
const { pathToFileURL } = require("url");
```

**To Reproduce**

See example [here](https://semgrep.dev/s/Nn5A) or above.

**Expected behavior**

No warnings should be reported.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Desktop (please complete the following information):**
 - OS: Windows 11 22H2 (and reproduced at semgrep playground)
 - Semgrep Version: 1.17.1

**Comment by mjambon** [comment]
Thank you!

Here's another illustration of the bug:
![image](https://user-images.githubusercontent.com/343265/231887225-80b059cc-6635-4996-a6f1-a0638a294c70.png)

There are two matches on line 2, the first one being wrong:
```
⬆Line 2
X is `const`, Y is `const {b} = 2`
⬆Line 2
X is `{b}`, Y is `2`
```

https://semgrep.dev/s/pOp0
